File: 2005-check-if-all-the-integers-in-a-range-are-covered/check-if-all-the-integers-in-a-range-are-covered.php
class Solution {

    /**
     * @param Integer[][] $ranges
     * @param Integer $left
     * @param Integer $right
     * @return Boolean
     */
    function isCovered($ranges, $left, $right) {
        
        for ($i = $left; $i <= $right; $i++){
            $covered = false;
            foreach($ranges as $x) {
                if (($x[0] <= $i) and ($i <= $x[1])){
                    $covered = true;
                    break;
                }
            }
            if (!$covered) return false;
        }
        return true;
    }
}

File: 2005-check-if-all-the-integers-in-a-range-are-covered/check-if-all-the-integers-in-a-range-are-covered.py
class Solution:
    def isCovered(self, ranges: List[List[int]], left: int, right: int) -> bool:

        for num in range(left , right + 1):
            for i,j in ranges :
                if i <= num <= j:
                    break
            else: # if we go through all the elements in range and if it doesnt break return false bc its not covered in the range 
                return False 
        return True
        
        